daml-script/daml/Daml/Script.daml

```diff
@@ -829,10 +822,13 @@ checkUserNotFound userId script = do
 
 -- | HIDE Convert the text to a user id or fail if it does not conform to the format restriction.
 validateUserId : HasCallStack => Text -> Script UserId
-validateUserId name = mkUserId name $ lift $ Free $ ValidateUserId ValidateUserIdPayload with
-  continue = pure
-  locations = getCallStack callStack
-  name
+validateUserId name =
+  case validateUserIdBuiltin name of
+    None -> pure (UserId name)
+    Some msg -> throw (InvalidUserId msg)
+
+validateUserIdBuiltin : Text -> Optional Text
```

Comment:
I assume you’ll expose the primitive in a separate PR?

Reply:
I'm a bit undecided here...

I don't think we should expose the raw builtin:
```
validateUserIdBuiltin : Text -> Optional Text
```
We might expose:
```
validateUserId : Text -> Either Text UserId -- Left is the error reason
```
But do we want this pure version *instead of* or *as well as* the version which runs in `Script`
```
validateUserId : HasCallStack => Text -> Script UserId -- may throw: InvalidUserId
```
If we ditch the `Script` version , we also get to ditch the `InvalidUserId` exception.
(In either case I can do it in a subsequent PR.)